src/components/Article.jsx

```diff
@@ -10,7 +10,16 @@ export const Article = () => {
     const [loading, setLoading] = useState(true)
     const [error, setError] = useState(false)
     const [errMsg, setErrMsg] = useState("Something went wrong, please navigate back to the Articles list and try again")
-    
+    const [userVote, setUserVote] = useState(0)
+    const [voteMessage, setVoteMessage] = useState("Click arrow to up vote or down vote")
+
+    useEffect(() => {
+        setUserVote(0)
+        setVoteMessage("Click arrow to up vote or down vote")
+    }, [window.location.hash]) //resets optimistic rendering when switching article via url
+
+
```

Comment:
I'm not sure what `window.location.hash` in the dependency array is having the effect you are thinking here. If you switch article via the URL, then your whole react app is unmounted and rendered from scratch (like a page refresh) so even if the dependency array is empty, it would still re-run that effect.

If there is a reason why this solves a problem that I have missed then keep it in, but I tried removing it and it seemed like I could navigate between articles no problem.

Reply:
There is no refresh on manually changing article number in url path because of the hashed routing.  

if you remove that window.location.hash dependency, then upvote twice (so message is displaying "you can only upvote once"), change the number and hit return, it will still show that message on the next articles page.
So you can navigate fine without it, but the votes will be wrong